java/src/jmri/jmrix/dccpp/DCCppPredefinedMeters.java

```diff
@@ -17,49 +24,96 @@
             }
         };
-        
-        currentMeter = new DefaultMeter.DefaultCurrentMeter(
-                memo.getSystemPrefix() + "V" + "BaseStationCurrent",
-                Meter.Unit.Percent, 0, 100.0, 1.0, updateTask);
-        
-        InstanceManager.getDefault(MeterManager.class).register(currentMeter);
-        
+
         // TODO: For now this is OK since the traffic controller
         // ignores filters and sends out all updates, but
         // at some point this will have to be customized.
-        tc.addDCCppListener(DCCppInterface.THROTTLE, this);
+        tc.addDCCppListener(DCCppInterface.CS_INFO, this);
 
-        //is_enabled = false;
         updateTask.initTimer();
-
-        log.debug("DCCppPredefinedMeters constructor called");
+        updateTask.enable(); //enable the send even if no meters exist yet
     }
 
     public void setDCCppTrafficController(DCCppTrafficController controller) {
         tc = controller;
     }
 
+    /* handle new Meter replies and original current replies 
+     *   creates meters if first time this name is encountered
+     *   builds original "current percent" meter or as defined in 
+     *   new MeterReply message format                               */
     @Override
     public void message(DCCppReply r) {
-        if (r.isCurrentReply()) {
-            log.debug("DCCppPredefinedMeters received reply: '{}'", r);
-            try {
-                currentMeter.setCommandedAnalogValue(((r.getCurrentInt() * 1.0f) / (DCCppConstants.MAX_CURRENT * 1.0f)) * 100.0f );  // return as percentage.
-            } catch (JmriException e) {
-                log.error("exception thrown when set current", e);
+
+        //bail if other message types received
+        if (!r.isCurrentReply() && !r.isMeterReply()) return; 
+        
+        log.debug("Handling reply: '{}'", r);
+
+        //assume old-style current message and default name and settings
+        String meterName = "CurrentPct";
+        double meterValue = 0.0;
+        String meterType = DCCppConstants.CURRENT;
+        Meter.Unit meterUnit = Meter.Unit.Percent;
+        double minValue = 0.0;
+        double maxValue = 100.0;
+        double resolution = 0.1;
+
+        //use settings from message if Meter reply
+        if (r.isMeterReply()) {
+            meterName = r.getMeterName();
+            meterValue= r.getMeterValue();
+            meterType = r.getMeterType();
+            minValue  = r.getMeterMinValue();
+            maxValue  = r.getMeterMaxValue();
+            resolution= r.getMeterResolution();
+            meterUnit = r.getMeterUnit();
+        }
+                
+        //create, store and register the meter if not yet defined
+        if (!meters.containsKey(meterName)) {
+            log.debug("Adding new meter '{}' of type '{}' with unit '{}'" , meterName, meterType, meterUnit);
+            Meter newMeter;
+            String sysName = systemPrefix + "V" + meterType + "_" + meterName; 
```

Comment:
It might be good to add a comment that "V" is for AnalogIO / Meter.

Reply:
I changed that to use MeterManager.typeLetter() instead of the literal. That should clarify it.